src/plugins/intel_gpu/src/plugin/transformations/fc_horizontal_fusion.cpp

```diff
@@ -0,0 +1,206 @@
+                    auto zp_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_convert->get_input_node_shared_ptr(0));
+                    scalar_zp_val = zp_const->cast_vector<int32_t>()[0];
+                }
+                fused_zps = zp_nodes[0];
+            }
+            if (is_scalar) {
+                for (size_t i = 1; i < zp_nodes.size(); ++i) {
+                    bool current_is_scalar = (ov::shape_size(zp_nodes[i]->get_output_shape(0)) == 1);
+                    if (!current_is_scalar)
+                        return false;
+                    // validate all zp values are same
+                    int32_t cur_zp_val = 0;
+                    if (auto zp_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_nodes[i])) {
+                        cur_zp_val = zp_const->cast_vector<int32_t>()[0];
+                    } else if (auto zp_convert = std::dynamic_pointer_cast<ov::op::v0::Convert>(zp_nodes[i])) {
+                        auto zp_const =
+                            std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_convert->get_input_node_shared_ptr(0));
+                        cur_zp_val = zp_const->cast_vector<int32_t>()[0];
+                    } else {
+                        OPENVINO_ASSERT("Unsupported zp input node for FC horizontal fusion");
+                    }
+                    if (cur_zp_val != scalar_zp_val)
+                        return false;
+                }
+            } else {
+                auto zp_nodes_as_output_vector = ov::OutputVector{zp_nodes[0], zp_nodes[1], zp_nodes[2]};
+                fused_zps = std::make_shared<ov::op::v0::Concat>(zp_nodes_as_output_vector, 0);
+                fused_zps->set_friendly_name(zp_nodes[0]->get_friendly_name() + "_fused");
+            }
+        }
+        // Create new fc with merged weights, bias, scale, zp
+        std::shared_ptr<ov::Node> new_fc;
+        if (fused_zps)
+            new_fc = std::make_shared<op::FullyConnectedCompressed>(input_node, fused_weight, fused_bias, fused_scale, fused_zps);
+        else
+            new_fc = std::make_shared<op::FullyConnectedCompressed>(input_node, fused_weight, fused_bias, fused_scale);
+
+        auto new_fc_name = fc_nodes[0]->get_friendly_name() + "_fused";
+        new_fc->set_friendly_name(new_fc_name);
+        copy_runtime_info(fc_nodes[0], new_fc);
+
+        // Split output and connect to the orig users
+        auto split_name = fc_nodes[0]->get_friendly_name() + "_split";
+        auto axis_const = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{1}, {new_fc->get_output_partial_shape(0).size() - 1});
+        auto split_const = ov::op::v0::Constant::create(ov::element::i64, ov::Shape{3}, orig_n_sizes);
+        auto output_split = std::make_shared<ov::op::v1::VariadicSplit>(new_fc, axis_const, split_const);
+        copy_runtime_info(fc_nodes[0], output_split);
```

Comment:
As above, It does not need to merge runtime info of the first fc into the newly created variadic split

Reply:
Same as above.